overlord/devicestate/devicestate_test.go

```diff
@@ -2794,3 +2815,531 @@ func (s *deviceMgrSuite) TestDeviceCtxProvided(c *C) {
 	c.Assert(err, IsNil)
 	c.Assert(deviceCtx1, Equals, deviceCtx)
 }
+
+var snapYaml = `
+name: foo-gadget
+type: gadget
+`
+
+var gadgetYaml = `
+volumes:
+  pc:
+    bootloader: grub
+`
+
+func mockSnapWithData(c *C, yamlText string, si *snap.SideInfo, content map[string]string) *snap.Info {
```

Comment:
This could go into snaptest.MockSnapWithFiles . We have snaptest.MockSnap and snaptest.MakeTestSnapWithFiles there already so this seems like a nice fit.

Reply:
I'll push that as a separate PR

Reply:
Pushed to #6918